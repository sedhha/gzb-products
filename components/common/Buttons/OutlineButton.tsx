import styled from 'styled-components';
type Props = {
  outlineColor: string;
  borderRadius?: string;
};

const OutlineButton = styled.button`
  outline: none;
  background-color: transparent;
  border: 2px solid ${(props: Props) => props.outlineColor};
  cursor: pointer;
  padding: 0.6rem 1rem;
  color: ${(props: Props) => props.outlineColor};
  border-radius: ${(props: Props) => props.borderRadius ?? '1rem'};
  font-size: 1rem;
  letter-spacing: 0.1rem;
  min-width: 10rem;
  transition: 0.3s ease all;
`;

export default styled(OutlineButton)`
  &:hover {
    background-color: ${(props: Props) => props.outlineColor};
    color: white;
  }
`;
