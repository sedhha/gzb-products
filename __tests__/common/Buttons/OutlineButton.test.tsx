import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import OutlineButton from '@common/Buttons/OutlineButton';

const outlineColor = 'red';
const borderRadius = '1rem';

describe('Outline Button Test Cases', () => {
  it('Outline Button should pick Outline Color dynamically: ', () => {
    const tree = renderer
      .create(
        <OutlineButton
          outlineColor={outlineColor}
          borderRadius={borderRadius}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', outlineColor);
    expect(tree).toHaveStyleRule('border-radius', borderRadius);
  });
});
