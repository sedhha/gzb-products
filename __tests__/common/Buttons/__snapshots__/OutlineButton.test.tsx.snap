// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outline Button Test Cases Outline Button should pick Outline Color dynamically:  1`] = `
.c0 {
  outline: none;
  background-color: transparent;
  border: 2px solid red;
  cursor: pointer;
  padding: 0.6rem 1rem;
  color: red;
  border-radius: 1rem;
  font-size: 1rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  min-width: 10rem;
  -webkit-transition: 0.3s ease all;
  transition: 0.3s ease all;
}

.c1:hover {
  background-color: red;
  color: white;
}

<button
  className="c0 c1"
/>
`;
